cranelift/filetests/src/function_runner.rs

```diff
@@ -246,8 +247,8 @@ impl UnboxedValues {
             ir::types::I16 => DataValue::I16(ptr::read(p as *const i16)),
             ir::types::I32 => DataValue::I32(ptr::read(p as *const i32)),
             ir::types::I64 => DataValue::I64(ptr::read(p as *const i64)),
-            ir::types::F32 => DataValue::F32(ptr::read(p as *const f32)),
-            ir::types::F64 => DataValue::F64(ptr::read(p as *const f64)),
+            ir::types::F32 => DataValue::F32(ptr::read(p as *const Ieee32)),
+            ir::types::F64 => DataValue::F64(ptr::read(p as *const Ieee64)),
```

Comment:
This is the only change (and similarly on lines 237-238) that might be a bit risky in this commit: these functions are used to store and load values for the trampoline into a compiled function and I could `repr(C)` both `Ieee32` and `Ieee64` if there are doubts about the [data layout](https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout).

Reply:
I added a `repr(C)` to `Ieee32` and `Ieee64` in the latest version of this commit; if they truly aren't necessary we can remove them later.